### PR TITLE
Views

### PR DESCRIPTION
**Pull request by AndreaM2429**
Hello there!
I added a new feature to the `BlogApp` project, so I appreciate you taking the time to review these new implementations, thanks in advance for all your comments and feedback.

# 🎯 Milestones:
📌 Implement the design from the sneak peek wireframes

📌 Use the methods created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md)

📌 Implement a way to navigate between pages

📌 To see some posts or comments, you might need to add some data, you can use the following or make your own:
```
one_user = User.create(name: 'Johannes Weber', photo: 'https://i.pravatar.cc/150?img=69', bio: 'Lorem ipsum dolor sit amet consectetur, adipisicing elit.Magni corporis quam maxime placeat aut numquam ea, illum atque repellendus veritatis vel optio illo itaque eligendi explicabo obcaecati maiores earum! Consequatur!')
two_user = User.create(name: 'Santiago Sánchez', photo: 'https://i.pravatar.cc/150?img=58', bio: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi itaque suscipit beatae ipsa magnam iusto incidunt quia soluta hic perspiciatis necessitatibus a nisi repellat, aliquid fugiat! Praesentium maxime eum magnam?')
three_user = User.create(name: 'Élise Moreau', photo: 'https://i.pravatar.cc/150?img=9', bio: 'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Dolores natus dignissimos repudiandae dolorem quos consequuntur? Eos repellendus quam, ea maiores vero nisi quia mollitia quod debitis expedita veritatis sequi vel.')
four_user = User.create(name: 'Alexey Fedorov', photo: 'https://i.pravatar.cc/150?img=3', bio: 'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Deleniti, reiciendis praesentium facilis dignissimos similique, officia illo modi deserunt aperiam esse, velit vero? Exercitationem obcaecati mollitia numquam sed odit commodi ipsa?')
five_user = User.create(name: 'Camila Oliveira', photo: 'https://i.pravatar.cc/150?img=16', bio: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Impedit nostrum laudantium ut possimus consequatur eum mollitia, magnam et quia quisquam eos fuga enim error vel minima reprehenderit expedita quos facere.')

one_post = Post.create(author: one_user, title: 'Hello', text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Nihil, dolores nesciunt inventore molestiae sapiente, velit libero ipsum, magni illo nisi quod ratione incidunt provident fuga expedita eligendi consequuntur distinctio asperiores.')
two_post = Post.create(author: one_user, title: 'Hi again', text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita reiciendis doloremque totam tempore, eum voluptatem mollitia magnam ullam! Quod saepe eaque odio perspiciatis at assumenda veniam quasi officia, necessitatibus soluta?')
three_post = Post.create(author: two_user, title: 'Hello', text: 'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Tempora vel accusamus, corporis et voluptatem quos dolor soluta officiis, aperiam asperiores modi amet cum alias quae placeat illum rem quidem aliquam.')
four_post = Post.create(author: three_user, title: 'Hello', text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Sapiente nostrum commodi, earum beatae veritatis quaerat iste aspernatur debitis? Fuga corrupti non facilis harum accusamus consequatur architecto expedita perspiciatis hic sequi!')
five_post = Post.create(author: three_user, title: 'Hello again', text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Alias, hic. Minus ipsa quidem nobis, accusantium, voluptatem itaque dolorem voluptas corrupti ratione nisi consequuntur voluptatibus distinctio vitae esse quaerat quam animi.')
six_post = Post.create(author: three_user, title: 'Glad to be here!', text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quia consequuntur inventore minus corporis consectetur. Nemo blanditiis eos repellendus rem architecto numquam reprehenderit aut enim sed, necessitatibus dolorem expedita molestias dolorum.')
seven_post = Post.create(author: three_user, title: 'Last update', text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Delectus sint placeat, iste illo consequuntur impedit minima incidunt, maxime eos facilis odio libero, assumenda pariatur magni cumque ipsam illum laboriosam. Numquam.')
eight_post = Post.create(author: three_user, title: 'Showing my work', text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam consequuntur iusto dolores ipsa, eaque possimus tenetur neque distinctio dignissimos magni eum quia aliquam ducimus dolore nostrum, modi, nemo non? Illum!')

Comment.create(post: one_post, author: five_user, text: 'Hello Johannes' )
Comment.create(post: one_post, author: five_user, text: 'Welcome' )
Comment.create(post: one_post, author: three_user, text: 'Hi there!' )
Comment.create(post: one_post, author: four_user, text: 'Nice to meet you!' )
Comment.create(post: one_post, author: two_user, text: 'Hi!' )
Comment.create(post: one_post, author: two_user, text: 'Nice to meet you!' )
Comment.create(post: six_post, author: first_user, text: 'Hi' )
Comment.create(post: six_post, author: five_user, text: 'Hello Élise' )
Comment.create(post: eight_post, author: first_user, text: 'Nice work' )
Comment.create(post: eight_post, author: three_user, text: 'I am impressed' )
Comment.create(post: eight_post, author: five_user, text: 'Awesome Élise' )
Comment.create(post: eight_post, author: two_user, text: 'That is an excellent work' )

Like.create(author: two_user, post: one_post)
Like.create(author: three_user, post: one_post)
Like.create(author: five_user, post: one_post)
Like.create(author: one_user, post: eight_post)
Like.create(author: two_user, post: eight_post)
Like.create(author: four_user, post: eight_post)
Like.create(author: five_user, post: eight_post)
``` 

# 🎖️ General requirements:
📍 No linter errors
📍 Gitflow
📍 Document work

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-  </head>
+<section class="user-container d-flex-col">
+  <%= render partial: 'users/user_card', object: @user, as: :user %>
 
-  <body>
-    <h2>List of posts, general views</h2>
-  </body>
-</html>
\ No newline at end of file
+  <%= render partial: 'posts_comments', collection: @posts, as: :post %>
```
Comment (afanimartin) [inline app/views/posts/index.html.erb:4]: - Please use the method `five_recent_comments` to render the recent comments for each post as required.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @AndreaM2429, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Users and Posts controller added with appropriate actions for showing user and post details
- Proper use of partials. Nice job!
- Project uses a professional readme template
- App works as expected

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (AndreaM2429) [reply app/views/posts/index.html.erb:4]: Hi @afanimartin, if you go to the `app/views/posts/_posts_comments`, I'm using a method named `comments_exist` that is in the `helpers/application_helper` where I'm using the method to render the recent comments.
Here you can see that the post has 6 comments but it only shows 5 as the requirements
image

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @AndreaM2429 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark: Project implementation method works as required 
:heavy_check_mark: Excellent communication with reviewer

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AndreaM2429** [comment]
@t-yanick thanks for your time

**Pull request merged**